File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        String[] str = logLine.split(":");
        return str[1].trim();
    }

    public static String logLevel(String logLine) {
        String[] str = logLine.split(":");
        String logWithBrackets = str[0].toLowerCase();
        return logWithBrackets.substring(1, logWithBrackets.length()-1);
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" +  logLevel(logLine) + ")";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {

    std::string delimiter = ": ";

    std::string message(std::string log_line){
        return log_line.substr(log_line.find(delimiter) + 2);
    }

    std::string log_level(std::string log_line){
        return log_line.substr(1, log_line.find(delimiter) - 2);
    }

    std::string reformat(std::string log_line){
        return message(log_line) + " (" + log_level(log_line) + ")";
    }
}
